### DISCUSSION TITLE
Differences between Copilot Chat VSCode and VS2022 are unexplainable

### DISCUSSION THREAD

**Discussion by Hefaistos68** [community/community#58173] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I am using copilot since about 2 months and am mostly satisfied, a few things here and there, some more of this and that, but I am sure this will come soon.
But what I dont get is the huge difference in functionality between the VScode version and the VS2022 version. The Visual Studio version almost seems to be a forgotten son compared to the features in the VScode version. And to add insult to injury, the VS2022 extension even gives you answers like that:
![image](https://github.com/community/community/assets/8855055/171f8eb0-60e2-4c51-8329-8c1c6f292196)

So I have to install VSCode to give feedback for the VS2022 version? Clearly someone is using shared code but not enough I´d say.

The VSCode version has so many more embedded features that are missing in VS2023, and when trying to use them one gets an explanation on how to use the feature on VSCode instead. see ->
First try:
![image](https://github.com/community/community/assets/8855055/850e32dc-936c-431d-92a5-a399755aad57)
Second try (seems its learning something at least):
![image](https://github.com/community/community/assets/8855055/530ae317-e146-4041-b196-65418e370bff)

What is interesting though is the output of the Copilot in the output window in Visual Studio, seems that with every request for "/doc" another prompt is created. It varies, but the last one is especially funny.
```
[Conversations Verbose] [KernelLogger] Rendered prompt: You are an AI programming assistant. When asked for you name, you must respond with "GitHub Copilot".
[Conversations Verbose] [KernelLogger] Rendered prompt: A software developer is using an AI chatbot in a code editor. 
[Conversations Verbose] [KernelLogger] Extracting blocks from template: A developer working in Visual Studio wants to automate their experience in Visual Studio. They've asked you to build them a plan to satisfy the goal "{{$goal}}". You are an expert software developer who can build logical, efficient and accurate plans to accomplish goals.
[Conversations Verbose] [KernelLogger] Your task is to generate an explanation for the Developer's ask of "{{$input}}", keep this in mind when reading over the below sections for context. Pay especially close attention to the selection, exception context, and the last messages from the previous conversation. You also have a background as a highly skilled professor specializing in software development.
```

**Comment by chrisonoo** [comment]
I also noticed a big discrepancy and it's a bit strange. So should I start coding ASP.NET Core apps in VS Code to have a better Copilot? A lot of time has passed since the thread was created and nothing has changed 🤔

**Reply by ayim** [reply]
Hey - VS Copilot team here. @rheapatel will have the latest details on chat prompting

**Comment by warchiefmarkus** [comment]
Yes! I use copilot everywhere I can, VS, VS Code, Android Studio, etc, and everywhere it has differences, not only chat, but code suggestions also. For example VS Code copilot code suggestions can now suggest code inside (in the middle of existing code) code string, but VS and AS versions can't, it suggests only on enter new string or some thing like this, but not inside existing string ! Looks like VS Code has most featureable and new copilot engine 🤔 (I use preview versions of Visual Studio 2022 and copilot so it must have latest changes).

**Reply by ayim** [reply]
With code suggestions in VS, we made the choice to not suggest in the middle of a string due to considerations with the brace completion experience.

Have you noticed the feature in VS where IntelliSense can steer the Copilot prediction for C#/C++/C?
https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Reply by warchiefmarkus** [reply]
Yes, I read VS changelists and turned on it when it was just released. 
Suggestions in the middle of the string are really helpful and cool in many situations, sometimes I just keep open a project in VS and VS Code at the same time to write some classes quickly.

MAYBE, if it does not need much time and changes to the existing version, try to add an option to turn off/on suggestions in the middle of the string in VS? I think there are a lot of people who will be interested in such option.

**Reply by ayim** [reply]
I'll check in w/ team and see what's possible; I'm impressed by your attention to detail

**Reply by warchiefmarkus** [reply]
Thanks, will be glad to hear about some decisions in that option! Visual Studio is really best and full of features IDE (for me, at least) for .NET/C++, and  it's interesting to explore each time the abilities bring new update :)
